lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -831,25 +830,39 @@ def make_mpiexec_hydra_options(self):
             else:
                 self.log.debug("make_mpiexec_hydra_options: no rmk from HYDRA_RMK %s and hydra_info %s",
                                self.HYDRA_RMK, self.hydra_info)
+
+        launcher = None
+        default_launcher = getattr(self, 'HYDRA_LAUNCHER', None)
+        avail_launchers = self.hydra_info.get('launcher', [])
+
+        if self.options.launcher:
+            launcher = self.options.launcher
+            if launcher not in avail_launchers:
+                err = "Specified launcher %s does not exist, available launchers: %s"
+                self.log.raiseException(err % (launcher, avail_launchers))
```

Comment:
Do we want to enforce only using known launcher, or just warn about using what looks like an unknown launcher and then continue anyway? I would do the latter...